store/sqlstore/thread_store.go

```diff
@@ -501,14 +501,17 @@ func (s *SqlThreadStore) DeleteMembershipForUser(userId string, postId string) e
 	return nil
 }
 
-func (s *SqlThreadStore) CreateMembershipIfNeeded(userId, postId string, following, incrementMentions, updateFollowing bool) error {
+func (s *SqlThreadStore) MaintainMembership(userId, postId string, following, incrementMentions, updateFollowing, updateViewedTimestamp bool) error {
 	membership, err := s.GetMembershipForUser(userId, postId)
 	now := utils.MillisFromTime(time.Now())
 	if err == nil {
-		if (updateFollowing && !membership.Following || membership.Following != following) || incrementMentions {
+		if (updateFollowing && !membership.Following || membership.Following != following) || incrementMentions || updateViewedTimestamp {
```

Comment:
This has gotten a bit hardcore. Can we add a comment explaining the logic?

Reply:
done